modules/subscriptions.php

```diff
@@ -25,6 +25,33 @@ function jetpack_subscriptions_configuration_load() {
 	exit;
 }
 
+/**
+ * Cherry picks keys from `$_SERVER` array.
+ *
+ * @since 6.0.0
+ *
+ * @return array An array of server data.
+ */
+function jetpack_subscriptions_cherry_pick_server_data() {
+	$data = array();
+
+	foreach ( $_SERVER as $key => $value ) {
+		if ( ! is_string( $value ) ) {
+			continue;
+		} elseif ( 0 === strpos( $key, 'HTTP_COOKIE' ) ) {
+			continue;
+		}
+
+		if ( 0 === strpos( $key, 'HTTP_' ) ) {
+			$data[ $key ] = $value; // HTTP headers.
+		} elseif ( in_array( $key, array( 'REMOTE_ADDR', 'REQUEST_URI', 'DOCUMENT_URI' ), true ) ) {
+			$data[ $key ] = $value;
+		}
```

Comment:
These `if-elseif` can be:

```php
if ( ! is_string( $value ) || 0 === strpos( $key, 'HTTP_COOKIE' ) ) {
	continue;
}

if ( 0 === strpos( $key, 'HTTP_' ) || in_array( $key, array( 'REMOTE_ADDR', 'REQUEST_URI', 'DOCUMENT_URI' ), true ) ) {
	$data[ $key ] = $value;
}
```

Reply:
Good point, this would make the code more DRY.

Reply:
Good suggestion. Fixed, thanks!